osquery/tables/system/windows/smbios_tables.cpp

```diff
@@ -33,6 +34,49 @@ std::string to_iso8601_date(const FILETIME& ft) {
   return iso_date.str();
 }
 
+std::string getFirmwareType() {
+  enum class FirmwareType : std::uint32_t {
+    Unknown,
+    Bios,
+    Uefi,
+  };
+  using GetFirmwareTypePtr = BOOL (*)(FirmwareType*);
+
+  auto kernel32_module = GetModuleHandleA("kernel32");
+  auto function_ptr = reinterpret_cast<GetFirmwareTypePtr>(
+      GetProcAddress(kernel32_module, "GetFirmwareType"));
+  FirmwareType firmware_type = FirmwareType::Unknown;
+
+  if (function_ptr == nullptr) {
+    // We are on Windows 7: Attempt to determine the firmware type based on
+    // the registry keys
+    HKEY state_reg_key;
+    auto success =
+        RegCreateKeyA(HKEY_LOCAL_MACHINE,
```

Comment:
Please don't use an API that can potentially write the registry.
We should only try to open it.
Use `RegOpenKeyExA` instead.

Reply:
Done. Thanks for catching this!